mantidproject/mantid
# Clean up and refactor Muon fitting code

**Description of work.**
This PR removes some duplicated code in the fitting tab, which emerged when the sequential fitting tab was created. It also corrects a few behaviours which improves usability:

1. Plot guess will now update when a parameter is changed
2. When a function parameter is changed in the fitting tab the sequential fitting tab will update accordingly
3. The fitting options (at the bottom of the fitting tab) will update the model when they are changed. Previously the options in the model would only update when the function structure was changed.

In addition, some new unit tests have been added for the fitting tab model and presenter.



**To test:**
1. Load in some data (MUSR62260)
2. Go to the fitting tab
3. Add a fit function (GausOsc)
4. Check the following (amongst a number of other things):
     1. Plot guess - the guess should update if a parameter is changed)
     2. Do a single fit 
     3. Go the sequential fitting tab, the parameter for one of the fits should have updated
     3. Change start and end times and redo the fit, the range should have changed.
     4. Change to simultaneous fit and do the same steps (ii-iv)
5. In general, play around with it a bit to see if anything breaks. 



Fixes #28531 . 




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
